### PR TITLE
Reviews for Sessions added

### LINKED ISSUE #179: Accept rating for talk
System should allow accepting rating & feedback for a session that has become over. 

See issue https://github.com/fossasia/open-event-webapp/issues/106

### PR DESCRIPTION
**Pull request by SaptakS**
I have added the database model, views and helper python functions so that it may function in the following way.

It is basically a corrected PR of #186 
Fixes #179

### CODE REVIEW

**Comment by SaptakS** [comment]
@leto @creativepsyco @rafalkowalski

**Comment by rafalkowalski** [comment]
@SaptakS Have you tested it?

**Comment by SaptakS** [comment]
@rafalkowalski its working fine for me. Please verify once.

File: open_event/helpers/data.py [hunk 1]
```diff
@@ -76,6 +76,42 @@ def remove_track(track_id):
         flash('You successfully deleted track')
 
     @staticmethod
+    def create_review(data, session_id, is_accepted=True):
+        """
+        review will be saved to database with proper Event id
```
Comment (rafalkowalski) [inline open_event/helpers/data.py]: i think it should be "proper session_id"

Comment (SaptakS) [inline open_event/helpers/data.py]: Yeah. Maybe the comments are a bit messed up.I will have to check.

File: migrations/versions/0b5368c41418_.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '0b5368c41418'
+down_revision = '202b15987640'
+
+from alembic import op
+import sqlalchemy as sa
+import sqlalchemy_utils
```
Comment (juslee) [inline migrations/versions/0b5368c41418_.py]: Codacy Issue found: [Unused import sqlalchemy_utils](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624767/issues/source?bid=3233864&fileBranchId=3233881#l15)

File: migrations/versions/0b5368c41418_.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def upgrade():
+    ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('session_reviews',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('email', sa.String(), nullable=False),
```
Comment (juslee) [inline migrations/versions/0b5368c41418_.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624767/issues/source?bid=3233864&fileBranchId=3233881#l22)

File: open_event/helpers/data.py [hunk 4]
```diff
@@ -76,6 +77,42 @@ def remove_track(track_id):
...
+        :param data: review data
+        :param session_id: review belongs to Event by event id
+        """
+        new_review = Review(email=data["email"],
+                              rating=data["rating"],
+                              comment=data["comment"],
+                              session_id=session_id)
+
```
Comment (juslee) [inline open_event/helpers/data.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624790/issues/source?bid=3233864&fileBranchId=3233881#l90)

File: open_event/helpers/data.py [hunk 5]
```diff
@@ -76,6 +77,42 @@ def remove_track(track_id):
...
+                              comment=data["comment"],
+                              session_id=session_id)
+        
+        save_to_db(new_review, "review saved")
+
+    @staticmethod
+    def update_review(data, review, session_id):
+        """
+        review will be updated in database
+        :param data: review data
```
Comment (juslee) [inline open_event/helpers/data.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624790/issues/source?bid=3233864&fileBranchId=3233881#l97)

File: open_event/views/views.py [hunk 6]
```diff
@@ -278,6 +282,44 @@ def generate_icalender_track(track_id):
...
+                 'detail': 'Review Updated'})
+
+    elif "email" in data and "rating" in data: 
+        if not re.match(r"[^@]+@[^@]+\.[^@]+", data["email"]):
+            return jsonify(
+                {'status': '400',
+                 'detail': 'Not valid data'})
+        DataManager.create_review(data, session_id)        
+        return jsonify(
+                {'status': '200',
```
Comment (juslee) [inline open_event/views/views.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624826/issues/source?bid=3233864&fileBranchId=3233881#l317)

File: open_event/views/views.py [hunk 7]
```diff
@@ -278,6 +282,44 @@ def generate_icalender_track(track_id):
...
+        DataManager.update_review(data, reviews[0], session_id)
+        return jsonify(
+                {'status': '200',
+                 'detail': 'Review Updated'})
+
+    elif "email" in data and "rating" in data: 
+        if not re.match(r"[^@]+@[^@]+\.[^@]+", data["email"]):
+            return jsonify(
+                {'status': '400',
+                 'detail': 'Not valid data'})
```
Comment (juslee) [inline open_event/views/views.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624826/issues/source?bid=3233864&fileBranchId=3233881#l314)

File: open_event/views/views.py [hunk 8]
```diff
@@ -278,6 +282,44 @@ def generate_icalender_track(track_id):
...
+    if "comment" not in data:
+        data["comment"] = ""
+    reviews = DataGetter.get_reviews_by_email_and_session_id(data["email"])
+
+    if reviews.count() > 0:
+        DataManager.update_review(data, reviews[0], session_id)
+        return jsonify(
+                {'status': '200',
+                 'detail': 'Review Updated'})
+
```
Comment (juslee) [inline open_event/views/views.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624826/issues/source?bid=3233864&fileBranchId=3233881#l309)

File: open_event/views/views.py [hunk 9]
```diff
@@ -9,8 +10,11 @@
 from ..models.microlocation import Microlocation
 from ..models.event import Event
 from ..models.session import Session, Level, Format, Language
+from ..models.reviews import Review
 from ..models.version import Version
 from ..helpers.object_formatter import ObjectFormatter
+from ..helpers.data import DataManager, save_to_db, delete_from_db
```
Comment (juslee) [inline open_event/views/views.py]: Codacy Issue found: [Unused save_to_db imported from helpers.data](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624826/issues/source?bid=3233864&fileBranchId=3233881#l16)

File: open_event/models/reviews.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+class Review(db.Model):
+    """Review model class"""
+    __tablename__ = 'session_reviews'
+    id = db.Column(db.Integer, primary_key=True)
+    email = db.Column(db.String, nullable=False)
+    comment = db.Column(db.String)
+    rating = db.Column(db.Integer, nullable=False)
+    session_id = db.Column(db.Integer, db.ForeignKey('session.id'))
+
```
Comment (juslee) [inline open_event/models/reviews.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624804/issues/source?bid=3233864&fileBranchId=3233881#l12)

File: open_event/models/reviews.py [hunk 11]
```diff
@@ -0,0 +1,33 @@
+from . import db
+from .session import Session
```
Comment (juslee) [inline open_event/models/reviews.py]: Codacy Issue found: [Unused Session imported from session](https://www.codacy.com/app/triplez/open-event-orga-server/file/2406624804/issues/source?bid=3233864&fileBranchId=3233881#l2)

File: open_event/helpers/data.py [hunk 12]
```diff
@@ -76,6 +77,42 @@ def remove_track(track_id):
         flash('You successfully deleted track')
 
     @staticmethod
+    def create_review(data, session_id, is_accepted=True):
```
Comment (juslee) [inline open_event/helpers/data.py]: Codacy Issue found: [Unused argument 'is_accepted'](https://www.codacy.com/app/triplez/open-event-orga-server/file/2411291172/issues/source?bid=3233864&fileBranchId=3233881#l80)

File: open_event/helpers/data.py (hunk 4 above)
Comment (juslee) [inline open_event/helpers/data.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2411291172/issues/source?bid=3233864&fileBranchId=3233881#l90)

File: open_event/helpers/data.py (hunk 5 above)
Comment (juslee) [inline open_event/helpers/data.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2411291172/issues/source?bid=3233864&fileBranchId=3233881#l97)

File: migrations/versions/0b5368c41418_.py (hunk 2 above)
Comment (juslee) [inline migrations/versions/0b5368c41418_.py]: Codacy Issue found: [Unused import sqlalchemy_utils](https://www.codacy.com/app/triplez/open-event-orga-server/file/2411291149/issues/source?bid=3233864&fileBranchId=3233881#l15)

File: migrations/versions/0b5368c41418_.py [hunk 13]
```diff
@@ -0,0 +1,35 @@
+"""empty message
```
Comment (juslee) [inline migrations/versions/0b5368c41418_.py]: Codacy Issue found: [Invalid module name "0b5368c41418_"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2411291149/issues/source?bid=3233864&fileBranchId=3233881#l1)

File: migrations/versions/0b5368c41418_.py (hunk 3 above)
Comment (juslee) [inline migrations/versions/0b5368c41418_.py]: Codacy Issue found: [Trailing whitespace](https://www.codacy.com/app/triplez/open-event-orga-server/file/2411291149/issues/source?bid=3233864&fileBranchId=3233881#l22)

File: migrations/versions/0b5368c41418_.py [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+
+Revision ID: 0b5368c41418
+Revises: 202b15987640
+Create Date: 2016-04-09 16:25:09.139313
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '0b5368c41418'
```
Comment (juslee) [inline migrations/versions/0b5368c41418_.py]: Codacy Issue found: [Invalid constant name "revision"](https://www.codacy.com/app/triplez/open-event-orga-server/file/2411291149/issues/source?bid=3233864&fileBranchId=3233881#l10)

**Pull request closed without merging**